### PR TITLE
Add reduce function

### LINKED ISSUE #2: Add Reduce function
API:

```go
input := []int{1,2,3,4,5}

accumulator := 0 // initialize accumulator
godash.Reduce(input, &accumulator, func(acc, each int) int {
    return acc + each
})

// accumulator should be 15
```

### PR DESCRIPTION
**Pull request by dineshba**
Fixes https://github.com/thecasualcoder/godash/issues/2

### CODE REVIEW

**Comment by aswinkarthik** [comment]
could you add a text and example documentation of a Word count example? A standard Reduce example.

File: reduce.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	if input.Kind() == reflect.Slice {
+		outputKind := output.Elem().Kind()
+		reducerFnType := reducer.Type()
+		if outputKind != reducerFnType.In(0).Kind() {
+			return fmt.Errorf("reducer function's first argument has to be the type of out")
+		}
+		if input.Type().Elem().Kind() != reducerFnType.In(1).Kind() {
+			return fmt.Errorf("reducer function's second argument has to be the type of element of input slice")
```
Comment (aswinkarthik) [inline reduce.go]: Could you enrich error message with what exact Type is expected similar to filter.go

File: reduce_test.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
+package godash_test
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/thecasualcoder/godash"
+	"strconv"
+	"testing"
+)
+
+func TestReduce(t *testing.T) {
```
Comment (aswinkarthik) [inline reduce_test.go:10]: Could you add an example for a test of Word Count

Comment (dineshba) [reply reduce_test.go:10]: Added

File: reduce.go (hunk 1 above)
Comment (dineshba) [reply reduce.go]: Enriched. Could you please review now?

**Pull request merged**